conference.js

```diff
@@ -629,47 +630,63 @@ export default {
+                    .catch(err => {
+                        if (requestedAudio && requestedVideo) {
+
+                            // Try audio only...
+                            audioAndVideoError = err;
 
-                        return (
-                            createLocalTracksF({ devices: [ 'audio' ] }, true));
-                    } else if (requestedAudio && !requestedVideo) {
+                            return (
+                                createLocalTracksF({ devices: [ 'audio' ] }, true));
+                        } else if (requestedAudio && !requestedVideo) {
+                            audioOnlyError = err;
+
+                            return [];
+                        } else if (requestedVideo && !requestedAudio) {
+                            videoOnlyError = err;
+
+                            return [];
+                        }
+                        logger.error('Should never happen');
+                    })
+                    .catch(err => {
+                        // Log this just in case...
+                        if (!requestedAudio) {
+                            logger.error('The impossible just happened', err);
+                        }
                         audioOnlyError = err;
 
-                        return [];
-                    } else if (requestedVideo && !requestedAudio) {
+                        // Try video only...
+                        return requestedVideo
+                            ? createLocalTracksF({ devices: [ 'video' ] }, true)
+                            : [];
+                    })
+                    .catch(err => {
+                        // Log this just in case...
+                        if (!requestedVideo) {
+                            logger.error('The impossible just happened', err);
+                        }
                         videoOnlyError = err;
 
                         return [];
-                    }
-                    logger.error('Should never happen');
-                })
-                .catch(err => {
-                    // Log this just in case...
-                    if (!requestedAudio) {
-                        logger.error('The impossible just happened', err);
-                    }
-                    audioOnlyError = err;
+                    });
 
-                    // Try video only...
-                    return requestedVideo
-                        ? createLocalTracksF({ devices: [ 'video' ] }, true)
-                        : [];
-                })
-                .catch(err => {
-                    // Log this just in case...
-                    if (!requestedVideo) {
-                        logger.error('The impossible just happened', err);
-                    }
-                    videoOnlyError = err;
+            tryCreateLocalTracks = this._videoBlurScriptPromise
+                .then(() =>
+                    _createLocalTracks([ getJitsiMeetGlobalNS().effects.JitsiStreamBlurEffect ])
```

Comment:
Let's add all the effects that live in `getJitsiMeetGlobalNS().effects`. Maybe you can do something like:
```
Object.values(getJitsiMeetGlobalNS().effects)
```

Reply:
Using Object.values(getJitsiMeetGlobalNS().effects) implies that we are using all effects available in the effects object but in this case we just want to load dynamically the video-blur-effect bundle and use only the JitsiStreamBlurEffect class if needed, right? So we are interested only in the blur effect.